libbeat/docs/config-file-format.asciidoc

```diff
@@ -0,0 +1,80 @@
+[[config-file-format]]
+== Config file format
+
+Beats config files are based on http://www.yaml.org[YAML], a file format that is easier to read and write then other common data format like XML or JSON.
+
+In beats all YAML files start with a "dictionary" (often called a "hash"). In addition to dictionaries, YAML also supports lists, numbers, strings, and many other data types. All members of lists and dictionaries must have the same indentation level.
+
+Dictionaries are represented by simple `key: value` pairs all having the same indentation level. The collon after `key` must be followed by a space.
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+name: John Doe
+age: 34
+country: Canada
+------------------------------------------------------------------------------
+
+Lists are introduced by dashes `- `. All list members will be lines beginning with `- ` at the same indentation level.
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+- Red
+- Green
+- Blue
+------------------------------------------------------------------------------
+
+Lists and dictionaries are used in beats to build structured configurations.
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+filebeat:
+  prospectors:
+    - input_type: log
+      paths:
+        - tmp/nasa/*.log
+      multiline:
+        pattern: ^\[
+        match: after
+------------------------------------------------------------------------------
+
+Lists and dictionaries can also be represented in abbreviated form. Abbreviated form is somewhat similar to JSON using `{}` for dictionaries and `[]` for lists:
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+person: {name: "John Doe", age: 34, country: "Canada"}
+colors: ["Red", "Green", "Blue"]
+------------------------------------------------------------------------------
+
+=== Enironment Variables
+
+=== Namespacing
+
+=== Config file Data Types
+
+==== Boolean
+
+==== Number
+
+==== String
+
+==== Duration
+
+==== Regular Expression
+
+==== Event Format String
+
+==== References
+
+=== YAML Tips and Gotchas
+
+[float]
+==== Use Spaces for Indentation
+
+Indentation is meaningful in YAML. Make sure that you use spaces, rather than tab characters, to indent sections. 
```

Comment:
We could add something like: "In the default configuration files and in all the examples in the documentation, we use 2 spaces per indentation level. We recommend you do the same."